packages/machine-id/binding.cc

```diff
@@ -0,0 +1,170 @@
+#include <napi.h>
+#include <string>
+
+#ifdef __APPLE__
+#include <CoreFoundation/CoreFoundation.h>
+#include <IOKit/IOKitLib.h>
+#elif defined(__linux__)
+#include <fstream>
+#include <algorithm>
+#elif defined(_WIN32)
+#include <windows.h>
+#endif
+
+using namespace Napi;
+
+namespace
+{
+
+#ifdef __APPLE__
+  // Get macOS machine ID using IOKit framework directly
+  std::string getMachineId()
+  {
+    std::string uuid;
+    io_registry_entry_t ioRegistryRoot = IORegistryEntryFromPath(kIOMainPortDefault, "IOService:/");
+
+    if (ioRegistryRoot == MACH_PORT_NULL)
+    {
+      return "";
+    }
+
+    CFStringRef uuidKey = CFSTR("IOPlatformUUID");
+    CFTypeRef uuidProperty = IORegistryEntryCreateCFProperty(ioRegistryRoot, uuidKey, kCFAllocatorDefault, 0);
+
+    if (uuidProperty)
+    {
+      char buffer[128];
+      if (CFStringGetCString((CFStringRef)uuidProperty, buffer, sizeof(buffer), kCFStringEncodingUTF8))
+      {
+        uuid = buffer;
+      }
+      CFRelease(uuidProperty);
+    }
+
+    IOObjectRelease(ioRegistryRoot);
+    return uuid;
+  }
+#elif defined(__linux__)
+  // Linux machine ID paths
+  const char *DBUS_PATH = "/var/lib/dbus/machine-id";
+  const char *DBUS_PATH_ETC = "/etc/machine-id";
+
+  // Trim whitespace and newlines from a string
+  std::string trim(const std::string &str)
+  {
+    if (str.empty())
+    {
+      return str;
+    }
+
+    std::string result = str;
+    result.erase(result.find_last_not_of(" \n\r\t") + 1);
```

Comment:
not sure if this is problematic / could end up in a weird state

Reply:
good catch, I'll explicitly handle that case just to make the behavior more intentionally documented